conmon-rs/server/src/container_log/journald.rs

```diff
@@ -0,0 +1,173 @@
+use crate::container_io::Pipe;
+use anyhow::{Context, Result};
+use getset::{CopyGetters, Getters, Setters};
+use std::{
+    marker::Unpin,
+    path::{Path, PathBuf},
+};
+use tokio::{
+    fs::{File, OpenOptions},
+    io::{AsyncBufRead, AsyncBufReadExt, AsyncWriteExt, BufReader, BufWriter},
+};
+use tracing::debug;
+
+#[derive(Debug, CopyGetters, Getters, Setters)]
+pub struct JournaldLogger {
+    #[getset(get)]
+    path: PathBuf,
+
+    #[getset(set)]
+    file: Option<BufWriter<File>>,
+
+    #[getset(get_copy)]
+    max_log_size: Option<usize>,
+
+    #[getset(get_copy, set)]
+    bytes_written: usize,
+}
+
+impl JournaldLogger {
+    const ERR_UNINITIALIZED: &'static str = "logger not initialized";
+
+    pub fn new<T: AsRef<Path>>(path: T, max_log_size: Option<usize>) -> Result<JournaldLogger> {
+        Ok(Self {
+            path: path.as_ref().into(),
+            file: None,
+            max_log_size,
+            bytes_written: 0,
+        })
+    }
+
+    pub async fn init(&mut self) -> Result<()> {
+        debug!("Initializing Journald logger in path {}", self.path().display());
+        self.set_file(Self::open(self.path()).await?.into());
```

Comment:
hm we shouldn't be writing to a file here, it should be the system log, similar to the journald server logger

Reply:
oh :)